sdk/src/keyed_account.rs

```diff
@@ -248,14 +249,20 @@ where
     }
 }
 
-pub fn from_keyed_account<S: Sysvar>(
-    keyed_account: &crate::keyed_account::KeyedAccount,
-) -> Result<S, InstructionError> {
+pub fn check_sysvar_keyed_account<'a, S: Sysvar>(
```

Comment:
Can't this be inlined at its only two sites?
Because otherwise this will be one more public interface that needs to be removed again,
when `KeyedAccounts` in general are removed.

Reply:
I don't see any problem with removing it then. I'd like to reuse the code here because consensus depends on the checks being the same.

Reply:
Yes, I thought you wanted to share it for the error messages to stay the same. However, you also intend to replace one of the call sites by the sysvar cache right? So, that would remove the potential error case there.

Reply:
This PR already changes the call site to use the sysvar cache, I'm not sure what you mean